src/caikit_nlp_client/http_client.py

```diff
@@ -292,3 +291,188 @@ def _create_json_request(self, model_id, text, **kwargs) -> dict[str, Any]:
+        json_input = {
+            "inputs": {
+                "source_sentences": source_sentences,
+                "sentences": sentences,
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/sentence-similarity-tasks",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response)
+
+    def rerank(
+        self, model_id: str, documents: list[dict[str, Any]], query: str, timeout: float = 60.0, parameters: dict[str, Any]={}
+    ) -> dict[str, Any]:
+
+        if not model_id:
+            raise ValueError("request must have a model id")
+
+        log.info(f"Calling rerank for '{model_id}'")
+        json_input = {
+            "inputs": {
+                "documents": documents,
+                "query": query,
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/rerank",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response)
+
+    def rerank_tasks(
+        self, model_id: str, documents: list[dict[str, Any]], queries: list[str], timeout: float = 60.0, parameters: dict[str, Any]={}
+    ) -> dict[str, Any]:
+
+        if not model_id:
+            raise ValueError("request must have a model id")
+
+        log.info(f"Calling rerank_tasks for '{model_id}'")
+        json_input = {
+            "inputs": {
+                "documents": documents,
+                "queries": queries,
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/rerank-tasks",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response)
+
+    def _unpack_or_raise_details(self, response: requests.Response, details_field=None) -> dict[str, Any]:
+        """
+        The details_field is present because there seems to be an inconsistent naming of this
+        field between different endpoints. Some have "details" others have "detail".
+
+        When details_field == None, try to detect the right name of the details field
+        """
+        if response.status_code == 200:
+            return response.json()
+
+        try:
+            if details_field is None:
```

Comment:
Can you add a comment pointing to https://github.com/caikit/caikit/issues/750?

Reply:
Added with [a0f761e](https://github.com/opendatahub-io/caikit-nlp-client/pull/135/commits/a0f761ea9da18dcaf4b52a2e571cb26044f78d59)